src/loki/matrix_action.cc

```diff
@@ -131,7 +130,10 @@ namespace valhalla {
 
       //check the distances
       auto max_location_distance = std::numeric_limits<float>::min();
-      check_distance(sources, targets, max_distance.find("sources_to_targets")->second, max_location_distance);
+      auto matrix_max_dist = max_matrix_distance.find(costing);
+      if (matrix_max_dist == max_matrix_distance.end())
```

Comment:
do you need this default thing here? it seems to me that it would be hiding a problem with the code. in other words you shouldnt be able to get to this point without a costing that should exist in this map right?